packages/neuron-ui/src/components/HistoryDetailPage/index.tsx

```diff
@@ -0,0 +1,277 @@
+      value:
+        (
+          <CopyZone content={transaction.hash} name={t('history.copy-tx-hash')}>
+            {transaction.hash}
+          </CopyZone>
+        ) || 'none',
+    },
+    {
+      label: t('transaction.block-number'),
+      value: transaction.blockNumber ? localNumberFormatter(transaction.blockNumber) : 'none',
+    },
+    {
+      label: t('transaction.date'),
+      value: +(transaction.timestamp || transaction.createdAt)
+        ? uniformTimeFormatter(+(transaction.timestamp || transaction.createdAt))
+        : 'none',
+    },
+    {
+      label: t('transaction.income'),
+      value: (
+        <CopyZone content={shannonToCKBFormatter(transaction.value, false, '')} name={t('history.copy-balance')}>
+          {`${shannonToCKBFormatter(transaction.value)} CKB`}
+        </CopyZone>
+      ),
+    },
+  ]
+
+  const inputsTitle = useMemo(
+    () => `${t('transaction.inputs')} (${transaction.inputs.length}/${localNumberFormatter(transaction.inputsCount)})`,
+    [transaction.inputs.length, transaction.inputsCount, t]
+  )
+
+  const outputsTitle = useMemo(() => {
+    return `${t('transaction.outputs')} (${transaction.outputs.length}/${localNumberFormatter(
+      transaction.outputsCount
+    )})`
+  }, [transaction.outputs.length, transaction.outputsCount, t])
+
+  const inputsData: InputOrOutputType[] = useMemo(
+    () => transaction?.inputs.map((item, idx) => ({ ...item, idx: idx + 1 })),
+    [transaction.inputs.length, transaction.inputsCount]
+  )
+  const outputsData: InputOrOutputType[] = useMemo(
+    () => transaction?.outputs.map((item, idx) => ({ ...item, idx: idx + 1 })),
+    [transaction.outputs.length, transaction.outputsCount]
+  )
+
+  const tabs = [
+    { label: inputsTitle, value: 0 },
+    { label: outputsTitle, value: 1 },
+  ]
+  const [currentTab, setCurrentTab] = useState<string | number>(tabs[0].value)
+
+  const handleTabChange = (e: number | string) => {
+    setCurrentTab(e)
+  }
+
+  const renderLockInfoDialog = useCallback(() => {
+    if (!lockInfo) {
+      return null
+    }
+    return <LockInfoDialog lockInfo={lockInfo} isMainnet={isMainnet} onDismiss={() => setLockInfo(null)} />
+  }, [lockInfo, isMainnet])
+
+  const handleListData = (cell: Readonly<State.DetailedInput | State.DetailedOutput>) => {
+    let address = ''
+    if (!cell.lock) {
+      address = t('transaction.cell-from-cellbase')
+    } else {
+      try {
+        address = scriptToAddress(cell.lock, isMainnet)
+      } catch (err) {
+        console.error(err)
+      }
+    }
+    const capacity = shannonToCKBFormatter(cell.capacity || '0')
+
+    return {
+      address,
+      capacity,
+    }
+  }
+
+  if (error.code) {
+    return (
+      <div className={styles.error}>
+        {error.message || t(`messages.codes.${ErrorCode.FieldNotFound}`, { fieldName: 'transaction' })}
+      </div>
+    )
+  }
+
+  const columns: TableProps<InputOrOutputType>['columns'] = [
```

Comment:
If the `columns` is not dependent on others' variables, it's better to define it out of the component.